include/libdnf/base/goal_elements.hpp

```diff
@@ -95,29 +99,46 @@ enum class GoalAction {
     RESOLVE
 };
 
+/// Settings for GoalJobSettings
 enum class GoalSetting { AUTO, SET_TRUE, SET_FALSE };
+
+/// Unresolved or resolved values based on GoalSetting
 enum class GoalUsedSetting { UNUSED, USED_TRUE, USED_FALSE };
 
 struct ResolveSpecSettings {
 public:
+    /// Important for queries that resolve SPEC
     bool ignore_case{false};
+    /// Important for queries that resolve SPEC
     bool with_nevra{true};
+    /// Important for queries that resolve SPEC
     bool with_provides{true};
+    /// Important for queries that resolve SPEC
     bool with_filenames{true};
+    /// Important for queries that resolve SPEC
     std::vector<libdnf::rpm::Nevra::Form> nevra_forms{};
 };
 
 struct GoalJobSettings : public ResolveSpecSettings {
 public:
+    /// Return used value for the strict
     GoalUsedSetting get_used_strict() const { return used_strict; };
+    /// Return used value for the best
     GoalUsedSetting get_used_best() const { return used_best; };
+    /// Return used value for the clean_requirements_on_remove
     GoalUsedSetting get_used_clean_requirements_on_remove() const { return used_clean_requirements_on_remove; };
```

Comment:
I would rephrase as "Return used value for strict" or "Return used values for the strict option". Same for best and clean_requirements_on_remove